### PR TITLE
Result aggregation

### LINKED ISSUE #26: Result aggregation & consolidation
In order to reduce FNs and FPs, we need to consolidate the results of multiple runs.

This means, that there are multiple things need to be done:
1. Results of multiple runs need to be combined
1.1. 'Majority voting': If an IND is detected in a majority of the aggregated runs, we can assume that it holds. Or rather if an IND is only detected in some cases, we have to do probabilization.
2. Detected INDs need to be probabilized with output of error-metric 'MissingValues' and statistics about the column.
2.1 We need to find a threshold/ratio between number of distinct values in the column and MissingValues in combination with the sample size, where we discard a detected IND

Note: This issue will be extended/refinded iteratively with the results of our experiments.

### PR DESCRIPTION
**Pull request by ClFeSc**
Closes #26

### CODE REVIEW

**Comment by ClFeSc** [comment]
This should now be ready for review. The immediate aim of these changes is to provide a framework for ranking INDs, not particularly providing the best possible ranking method.

**Comment by yjojo17** [comment]
Do you also get this warning message, when running the evaluation?
/home/yjojo/PycharmProjects/pythonProject/venv/lib/python3.10/lib/python3.10/site-packages/dataclasses_json/mm.py:271: UserWarning: Unknown type <class 'pysrc.models.errors.ErrorMetric'> at IND.errors: list[pysrc.models.errors.ErrorMetric] It's advised to pass the correct marshmallow type to `mm_field`.
  warnings.warn(

File: pysrc/scripts/evaluation.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import argparse
 import csv
+from dataclasses import dataclass
 import json
 import os
 import sys
 from typing import Literal, Optional
 
 import pandas as pd
+from pysrc.models.ind import IND, RankedIND
```
Comment (yjojo17) [inline pysrc/scripts/evaluation.py:10]: Maybe we can use relative imports here aswell to be consistent with the rest of the project? but it's only a suggestion

File: pysrc/scripts/sampling.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 from pathlib import Path
 
 from collections import defaultdict
+
+from pysrc.utils.is_non_zero_file import is_non_zero_file
```
Comment (yjojo17) [inline pysrc/scripts/sampling.py:15]: same here :)

**Review by yjojo17 — APPROVED** [review]
But I guess everything I mentioned is not blocking tho. And I haven't anything to add for the rest.

**Comment by ClFeSc** [comment]
> Do you also get this warning message, when running the evaluation? /home/yjojo/PycharmProjects/pythonProject/venv/lib/python3.10/lib/python3.10/site-packages/dataclasses_json/mm.py:271: UserWarning: Unknown type <class 'pysrc.models.errors.ErrorMetric'> at IND.errors: list[pysrc.models.errors.ErrorMetric] It's advised to pass the correct marshmallow type to `mm_field`. warnings.warn(

Yes, I get the same error. I suppose it's because we use a `list[ErrorMetric]` but only provide subclasses of `ErrorMetric` in that list. I don't know what to do about it, though. (So I'll do nothing about it I guess...)

File: pysrc/scripts/evaluation.py (hunk 1 above)
Comment (ClFeSc) [reply pysrc/scripts/evaluation.py:10]: I've accidentally pushed the changes for this to #37, so they'll be introduced there.

File: pysrc/scripts/sampling.py (hunk 2 above)
Comment (ClFeSc) [reply pysrc/scripts/sampling.py:15]: See above.

**Pull request merged**